crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -256,6 +303,60 @@ impl DataProvider for SuiClient {
     }
 }
 
+pub(crate) async fn convert_json_rpc_checkpoint(
+    cl: &SuiClient,
+    c: &sui_json_rpc_types::Checkpoint,
+) -> Result<Checkpoint> {
+    let digest = c.digest.to_string();
+    let sequence_number = c.sequence_number;
+
+    let validator_signature = c.validator_signature.encode_base64();
+    let validator_signature = Some(Base64::from(validator_signature.into_bytes()));
+
+    let previous_checkpoint_digest = c.previous_digest.map(|x| x.to_string());
+    let network_total_transactions = Some(c.network_total_transactions);
+    let rolling_gas_summary = convert_to_gas_cost_summary(&c.epoch_rolling_gas_cost_summary)
+        .await
+        .ok();
+    let epoch = Some(convert_to_epoch(cl, &c.epoch_rolling_gas_cost_summary).await?);
+
+    let end_of_epoch_data = &c.end_of_epoch_data;
+    let end_of_epoch = end_of_epoch_data.clone().map(|e| {
+        let committees = e.next_epoch_committee;
+
+        let new_committee = if committees.is_empty() {
+            None
+        } else {
+            Some(
+                committees
+                    .iter()
+                    .map(|c| CommitteeMember {
+                        authority_name: Some(c.0.into_concise().to_string()),
+                        stake_unit: Some(c.1),
+                    })
+                    .collect::<Vec<_>>(),
+            )
+        };
+
+        EndOfEpochData {
+            new_committee,
+            next_protocol_version: Some(e.next_epoch_protocol_version.as_u64()),
+        }
+    });
+
+    Ok(Checkpoint {
+        digest,
+        sequence_number,
+        validator_signature,
+        previous_checkpoint_digest,
+        live_object_set_digest: None, // TODO fix this
```

Comment:
I think this is the digest of ObjectRef = objectId, version, digest, iirc the digest of the transaction that modified the object 

Does that mean we're no longer returning all transactions of a checkpoint (as in rpc 1.0) or that the set of digests of live objects and the set of transaction digests are the same

Reply:
The live object set digest is a hash of the ObjectRefs of all objects that are live at the end of the checkpoint -- it's a kind of attestation, so that a fullnode can make sure that the objects that it has at the end of the checkpoint match what the network committed to.  It doesn't have anything to do with transactions.

Reply:
What do you mean by objects that are live? I am also quite confused about this. A checkpoint seems to have a list of `CheckpointCommitment`, but in the schema we have a `liveObjectSetDigest` of type string for the `Checkpoint` type.

Reply:
> A checkpoint seems to have a list of `CheckpointCommitment`, but in the schema we have a `liveObjectSetDigest` of type string for the Checkpoint type.

Yes the structure here is a little different between the data on-chain and GraphQL -- instead of having a list of commitments, the commitments are inlined into the `Checkpoint` so people can query for them individually (and if we don't have a particular commitment, we'll just return `None`).

> What do you mean by objects that are live?

An object reference is considered live if it's possible to use it as a transaction input (it has not already been used as a transaction input, deleted, or wrapped).  The live object set is the set of live object references and the `liveObjectSetDigest` commits to a particular set of live objects being live at the end of the checkpoint.

(You can say an object is considered live if it has a live object reference, but the more interesting property for the purposes of this digest is liveness of an object **reference**).

Reply:
Thanks for the explanation. I am still trying to wrap my head around it. For example, in the normal type `Checkpoint` has a list of `CheckpointCommittments`, and each `CheckpointCommittment` has a live digest `ECMHLiveObjectSetDigest`. I am not following how do I compute one liveObjectSetDigest from the set of `CheckpointCommittments`. I will follow up on this field because it's not clear to me how to implement it.